### ISSUE TITLE
PDF builds of French projects have issues with XeTeX from TL2016 or 2017

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#4272]
### Problem

- the polyglossia French module has not been modified upstream to adapt to internal changes at xetex 0.99994 or later, in practice this means for people using TeXLive 2016 (it has xetex 0.99996 whereas TL2015 has xetex 0.99992) or derived TeX distributions, or TeXLive 2017.

- for reference on this, see http://tug.org/pipermail/xetex/2017-December/027215.html

- some expected typographical rules of French are broken by this.

#### Procedure to reproduce the problem

file `index.rst`:

```rst
Bienvenue
=========

Un point-virgule ; doit être précédé par une espace fine typographique.

Un point-virgule; doit être précédé par une espace fine typographique.
```

in `conf.py`:

```
language = 'fr'
latex_engine = 'xelatex'
```

Then do `make latexpdf`.

#### Error logs / results

With TeXLive 2016 (`XeTeX 3.14159265-2.6-0.99996 (TeX Live 2016)`) PDF looks like this:

![capture d ecran 2017-12-03 a 14 34 21](https://user-images.githubusercontent.com/2589111/33525906-6b7b71da-d838-11e7-9051-a4dea08619b9.png)

The two lines should be identical; but the top one, whose source had a space before the `;` is wrong. It uses a too big space in output.

#### Expected results

Here is with TeXLive 2013's latex packages and xetex binary (`XeTeX 3.1415926-2.5-0.9999.3-2013061817 (TeX Live 2013)`)

![capture d ecran 2017-12-03 a 14 33 24](https://user-images.githubusercontent.com/2589111/33525904-5971520c-d838-11e7-965f-b2d9fe981f33.png)

As is expected and correct the two lines are identical with a thin space before the semi-colon.

### Environment info
- OS: Mac
- Python version: 3.5.4
- Sphinx version: 1.6.6+ (stable branch)
- TeXLive 2013 for good result, TeXLive 2016 for bad result.

### How to fix ?

Possibly, Sphinx should use babel + french and not polyglossia + french when latex engine is `xelatex` or `lualatex`.

Memo:

- independently of this polyglossia-french bug, the babel-french has more extensive support of French typographical rules, in particular for lists -- this babel-french customization is lacking in some aspects, but it does comply rather closely to official typography whereas polyglossia-french does _nothing at all_ in this area.

- babel-french is maintained, but polyglossia-french obviously much less though because the typographical issue reported here is consecutive to a change in XeTeX from 2016, and we are soon in 2018 and the problem is there,

- the #3546 issue shows,

  *  if using polyglossia: with TeXLive 2015 or earlier, that means XeTeX 0.99992 or earlier. It may also show with XeTeX prior to 0.99996, but 0.99996 is the first version I can easily test (TeXLive 2016) and which avoids the #3546 issue.

  * if using babel: with TeXLive 2014 or earlier, that means XeTeX 0.99991 or earlier. In particular the issue is avoided with TeXLive 2015, 16, 17.

   To recap: if Sphinx uses babel for French with latex_engine set to `xelatex`, the #3546 issue is avoided with XeTeX 0.99992 or later, whereas with polyglossia, it fails with 0.99992 and the first known (to me) valid XeTeX is XeTeX 0.99996. (but some earlier one might work).

**Comment by jfbu** [comment]
Upstream issue: https://github.com/reutenauer/polyglossia/issues/145

**Comment by jfbu** [comment]
Actually the upstream issue https://github.com/reutenauer/polyglossia/issues/145 is fixed upstream by merged PR https://github.com/reutenauer/polyglossia/pull/178, but

- the issue is still marked "Open",

- the polyglossia v1.42.5 release which will contain the merge is still to be finalized and pushed to CTAN and TeX distributions.

Anyway, this means that Linux TeX distros from years 2016--2017 at least will have that problem in future.

**Comment by jfbu** [comment]
Fixed at https://github.com/sphinx-doc/sphinx/commit/3156981d0905e1d9d31b36be8f9a4cff222f229d by using babel+French and not polyglossia+French with xelatex engine.
